addons/auth_signup/models/res_users.py

```diff
@@ -151,15 +151,23 @@ def _create_user_from_template(self, values):
             # copy may failed if asked login is not available.
             raise SignupError(ustr(e))
 
+    @api.model
+    def _get_reset_password_users(self, login):
+        users = self.search([('login', '=', login)])
+        if not users:
+            users = self.search([('email', '=', login)])
+        return users
+
     def reset_password(self, login):
         """ retrieve the user corresponding to login (login or email),
             and reset their password
         """
-        users = self.search([('login', '=', login)])
```

Comment:
Why not just doing 
```py
users = self.search(self._get_login_domain(login))
```
And create a `_get_email_domain(password)` that you would create next to `_get_login_domain` and override the same way in website?
Instead of the whole new `_get_reset_password_users` method

Reply:
Indeed it's way better. I didn't see that `_get_login_domain` sorry.

Reply:
No worries, it's why there is a review stage :)